src/nimblepkg/cli.nim

```diff
@@ -146,19 +146,43 @@ proc prompt*(forcePrompts: ForcePrompt, question: string): bool =
     else:
       return false
 
+proc promptCustom*(forcePrompts: ForcePrompt, question, default: string): string =
+  case forcePrompts:
+  of forcePromptYes:
+    display("Prompt:", question & " -> [forced " & default & "]", Warning,
+      HighPriority)
+    return default
+  else:
+    if default == "":
+      display("Prompt:", question, Warning, HighPriority)
+      displayCategory("Answer:", Warning, HighPriority)
+      let user = stdin.readLine()
+      if user.len == 0: return promptCustom(forcePrompts, question, default)
+      else: return user
+    else:
+      display("Prompt:", question & " [" & default & "]", Warning, HighPriority)
+      displayCategory("Answer:", Warning, HighPriority)
+      let user = stdin.readLine()
+      if user == "": return default
+      else: return user
+
 proc promptCustom*(question, default: string): string =
-  if default == "":
-    display("Prompt:", question, Warning, HighPriority)
-    displayCategory("Answer:", Warning, HighPriority)
-    let user = stdin.readLine()
-    if user.len == 0: return promptCustom(question, default)
-    else: return user
+  return promptCustom(dontForcePrompt, question, default)
+
+proc promptList*(forcePrompts: ForcePrompt, question: string, args: openarray[string]): string =
```

Comment:
This is great but what I had in mind was being able to make a selection using the arrow keys. I guess you might implement that later though.